sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TypeCoercion.scala

```diff
@@ -649,6 +650,21 @@ object TypeCoercion {
   }
 
   /**
+   * Coerces NullTypes in the Stack expression to the column types of the corresponding positions.
+   */
+  object StackCoercion extends Rule[LogicalPlan] {
+    def apply(plan: LogicalPlan): LogicalPlan = plan resolveExpressions {
+      case s @ Stack(children) if s.childrenResolved && s.hasFoldableRowNums =>
+        Stack(children.zipWithIndex.map {
```

Comment:
`children.tail.zipWithIndex.map { ...}`? So we can strip the first case and we don't need do `index - 1` in next case.